static/js/http.js

```diff
@@ -12,14 +12,16 @@ function request (method, path, options = {}) {
     headers.append('Content-Type', 'application/json')
     opts.body = JSON.stringify(body)
   }
-  return new Promise((resolv, reject) => {
-    window.fetch(path, opts)
-      .then(response => {
-        if (!response.ok)
-          return response.json().then(json => reject(new HTTPError(response.status, json.reason)))
-        return resolv(response.json())
-      })
-  })
+  return window.fetch(path, opts)
+    .then(response => {
+      if (!response.ok) {
+        return response.json().then(json => { throw new HTTPError(response.status, json.reason) })
```

Comment:
pretty sure i got "uncaught exception" when throwing inside the `then` callback, for some reason, maybe i did it differently, but that's why i introduced the extra `Promise`.. have you tested when eg. 403 is raised? logging in as non admin.

Reply:
I think it worked, but maybe not..? Maybe missed a case? I'll verify this.

Reply:
It seems to work just fine